### PR TITLE
Add SQL style guide

### PR DESCRIPTION
**Pull request by lnagle**

### CODE REVIEW

File: lang/sql/index.md [hunk 1]
```diff
@@ -0,0 +1,54 @@
+---
+title: Style Guide for SQL
+pageid: lang-sql
```
Comment (fidian) [inline lang/sql/index.md:3]: The yaml frontmatter needs a line of hyphens after it, just like the first line.

File: lang/sql/index.md [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+Keywords
+--------
+
+All keywords should be uppercase.
+
+    SELECT COUNT(*)
+    FROM someTable
+    WHERE DATEDIFF(NOW(), someTable.timeStamp)
+    ORDER BY ASC
```
Comment (fidian) [inline lang/sql/index.md]: You need to have a field name before ASC.

File: lang/sql/index.md [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+
+Keywords
+--------
+
+All keywords should be uppercase.
+
+    SELECT COUNT(*)
+    FROM someTable
+    WHERE DATEDIFF(NOW(), someTable.timeStamp)
```
Comment (fidian) [inline lang/sql/index.md]: For a more brief example, I would omit `someTable.` from this line.

File: lang/sql/index.md [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+    SELECT *
+    FROM someTable
+    WHERE column1 = true
+        AND column2 = "someValue"
+        AND column3 = false
+
+IN, AS, and OR should all be placed on the same line as the clause they are a part of.
+
+    SELECT *
+    FROM someTable
```
Comment (fidian) [inline lang/sql/index.md]: Add the AS here.

```
FROM someTable as st
```

File: lang/sql/index.md [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+    WHERE column IN config.columns
```
Comment (fidian) [inline lang/sql/index.md]: config.columns is not valid for this example.  

```
WHERE column IN ("column1", "column2")
```

File: lang/sql/index.md [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+
+The above rules apply for all subqueries as well.
```
Comment (fidian) [inline lang/sql/index.md]: Add a note about subqueries starting on a new line and also being indented further.

File: lang/sql/index.md [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+
+    SELECT *
+    FROM (
+        SELECT someTable.id AS someAlias, someTable.exampleColumn
```
Comment (fidian) [inline lang/sql/index.md]: Unless you're joining, having the table name before the fields only adds to the amount of data that you need to process.  Simpler SQL queries are better.

**Review by fidian — CHANGES_REQUESTED** [review]
Do queries start on their own line?  Do they have a trailing newline?  For instance, what would happen to the queries that you've embedded?  I think I would prefer that they are offset from non-SQL code.  Your thoughts?  Thoughts of the others?

```
return db->query("
    SELECT *
    FROM theTable
");
```

File: lang/sql/index.md [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+Indentation and Whitespace
+--------------------------
+
+Clauses should be separated by newlines.
+
+    SELECT *
+    FROM someTable
+    ORDER BY DESC
+
+When SELECTing multiple columns, and place on the same line.
```
Comment (AbsentSemicolon) [inline lang/sql/index.md]: I don't like laying things like this. Should just say "selecting".

File: lang/sql/index.md [hunk 9]
```diff
@@ -0,0 +1,54 @@
...
+
+    SELECT this, that, and, the, other
+    FROM someTable
+
+AND should be dropped onto a newline and given an additional indentation of four spaces relative to the clause it's a part of.
```
Comment (AbsentSemicolon) [inline lang/sql/index.md]: Same as above, I wouldn't call out what it is with it in all caps.

File: lang/sql/index.md (hunk 8 above)
Comment (lnagle) [reply lang/sql/index.md]: I agree with the quotes, but I think it should remain capitalized, as that is how it is used.

File: lang/sql/index.md (hunk 7 above)
Comment (lnagle) [reply lang/sql/index.md]: Fixed.

File: lang/sql/index.md (hunk 6 above)
Comment (lnagle) [reply lang/sql/index.md]: Added.

File: lang/sql/index.md (hunk 5 above)
Comment (lnagle) [reply lang/sql/index.md]: Fixed.

File: lang/sql/index.md (hunk 4 above)
Comment (lnagle) [reply lang/sql/index.md]: Fixed.

File: lang/sql/index.md (hunk 2 above)
Comment (lnagle) [reply lang/sql/index.md]: Fixed.

File: lang/sql/index.md (hunk 1 above)
Comment (lnagle) [reply lang/sql/index.md:3]: Fixed.

**Comment by lnagle** [comment]
Added an example for SQL queries embedded in code to this upcoming

File: lang/sql/index.md [hunk 10]
```diff
@@ -39,14 +39,21 @@ AND should be dropped onto a newline and given an additional indentation of four
 IN, AS, and OR should all be placed on the same line as the clause they are a part of.
```
Comment (AbsentSemicolon) [inline lang/sql/index.md]: These should be quoted as well, thought you'd see these too, but I should have called them out.

File: lang/sql/index.md [hunk 11]
```diff
@@ -0,0 +1,61 @@
+---
+title: Style Guide for SQL
+pageid: lang-sql
```
Comment (fidian) [inline lang/sql/index.md:3]: The keywords are missing.  Run `bin/update-all-keywords`

File: lang/sql/index.md [hunk 12]
```diff
@@ -0,0 +1,61 @@
...
+---
+
+
+Keywords
+--------
+
+All keywords should be uppercase.
+
+    SELECT COUNT(*)
+    WHERE DATEDIFF(NOW(), someTable.timeStamp)
```
Comment (fidian) [inline lang/sql/index.md]: Missing `FROM someTable` and I suggest removing `someTable.` from this line.

File: lang/sql/index.md [hunk 13]
```diff
@@ -0,0 +1,61 @@
...
+
+The above rules apply for all subqueries as well. Subqueries should also be started on a newline and indented an additional for space.
+
+    SELECT *
+    FROM (
+        SELECT id AS someAlias, exampleColumn
+        FROM someTable
+        LIMIT 3
+    )
+    ORDER BY DESC
```
Comment (fidian) [inline lang/sql/index.md]: Missing a field before `DESC`.

File: lang/sql/index.md [hunk 14]
```diff
@@ -0,0 +1,61 @@
...
+Indentation and Whitespace
+--------------------------
+
+Clauses should be separated by newlines.
+
+    SELECT *
+    FROM someTable
+    ORDER BY someField DESC
+
+When using "SELECT" on multiple columns, place the columns on the same line.
```
Comment (fidian) [inline lang/sql/index.md]: I think backticks are more appropriate for quotes here and for the next paragraph.  You're citing the code.

File: lang/sql/index.md [hunk 15]
```diff
@@ -0,0 +1,61 @@
...
+
+"AND" should be dropped onto a newline and given an additional indentation of four spaces relative to the clause it's a part of.
+
+    SELECT *
+    FROM someTable
+    WHERE column1 = true
+        AND column2 = "someValue"
+        AND column3 = false
+
+IN, AS, and OR should all be placed on the same line as the clause they are a part of.
```
Comment (fidian) [inline lang/sql/index.md]: These should also be surrounded with quotes or backticks.

**Review by fidian — CHANGES_REQUESTED** [review]
Couple more things.

File: lang/sql/index.md (hunk 12 above)
Comment (lnagle) [reply lang/sql/index.md]: Fixed.

File: lang/sql/index.md (hunk 14 above)
Comment (lnagle) [reply lang/sql/index.md]: Agreed, and changed.

File: lang/sql/index.md (hunk 15 above)
Comment (lnagle) [reply lang/sql/index.md]: Changed.

File: lang/sql/index.md (hunk 13 above)
Comment (lnagle) [reply lang/sql/index.md]: Fixed.

File: lang/sql/index.md (hunk 11 above)
Comment (lnagle) [reply lang/sql/index.md:3]: My version of BSD version of sed doesn't play nice with that script, and I'm not having much luck troubleshooting :(

Comment (lnagle) [reply lang/sql/index.md:3]: Nevermind, Derek and I made it work. Fixed.

**Review by fidian — APPROVED** [review]

**Comment by AbsentSemicolon** [comment]
Reviewed and looks good to me.

**Pull request merged**